### DISCUSSION TITLE
Connecting a NestJS commander CLI project to a server project

### DISCUSSION THREAD

**Discussion by minericpark** [jmcdo29/nest-commander#23] [Q&A]
Hello!

I'll break down the concept. I have two projects currently in a **monorepo**:
- main
  - packages
    - server
    - cli

I was working on developing a **CLI** that could **call various services** from a **separate server project** I've prototyped using **NestJS**. I was wondering how I would be able to **connect them together**? Would I have to use **manual GraphQL/REST calls** from the CLI application, **or** how would I **run both** an instance of the **server with the CLI**, then **access the server services**?

I've looked at the ogma/cli package, though it doesn't seem like it's calling any services from modules outside of that project, and haven't found any other examples and resources successfully :(

**Comment by jmcdo29** [comment]
I don't know of a way to use the _current_ instance that a server has, as that would be a separate process. What you could do is create libraries for the services you want to be able to access in both, and then the CLI can re-use the same service code the server has. Otherwise, you'll probably need to make GQL/HTTP calls to the server. Interesting use case though, I'll see if I can figure something out

**Reply by minericpark** [reply]
I'll try that, thanks!
